### PR TITLE
BXMSDOC-4153-master: Add process instance migration section

### PR DESCRIPTION
**Pull request by csherrar**

### CODE REVIEW

File: doc-content/enterprise-only/processes/process-instance-migration-con.adoc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[id='process-instance-migration-con']
+= Process instance migration
+
+Process instance migration (PIM) is a Technology Preview standalone service containing a user interface and a backend, and packaged as a Thorntail uber-jar. You can use the process instance migration service to define the migration between two different process definitions, known as a migration plan. The user can then apply the migration plan to the running process instance in a specific process server.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-con.adoc]: Thanks, backend is correct and Jar should be capitalized, as in uber-JAR.

File: doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
+[id='process-instance-migration-creating-plan-proc']
+= Creating a migration plan
+
+You can use the process instance migration service to define the migration between two different process definitions, known as a migration plan.
+
+.Prerequisites
+* You have defined processes in a backuped {PRODUCT} development environment.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc]: Done

File: doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+* The process instance migration service is running.
+
+.Procedure
+. Open the following address in a web browser:
++
+----
+localhost:3000
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc]: Done

File: doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,48 @@
+[id='process-instance-migration-installing-service-proc']
+= Installing the process instance migration service
+
+You can test the process instance migration service by cloning the GitHub repo, running the service, and accessing the service bin a browser.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc]: Done

[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc]: Done

File: doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
++
+[NOTE]
+====
+If required, install `npm`:
++
+----
+$ npm install
+----
+====
+. Once the service is up and running, open the following address in a web browser:
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc]: Done

File: doc-content/enterprise-only/processes/process-instance-migration-executing-plan-proc.adoc [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+. On the *Process Instance Migration* page, select the *Execute Migration Plan* image:processes/pim-execute-icon.png[] icon on the row of the migration plan you want to execute. The *Execute Migration Plan Wizard* window opens.
+. From the migration plan table, select the check box on the row of each running process instance you want to migrate, and click *Next*.
+. In the *Callback URL* field, enter the callback URL.
+. To the right of *Run migration*, perform one of the following tasks:
+* To execute the migration immediately, select *Now*.
+* To schedule the migration, select *Schedule* and in the text field, enter the date and time, for example `06/20/2019 10:00 PM`.
+. Click *Next*.
+. Optional: To export as a `JSON` file, click *Export*.
+. Click *Execute Plan*.
+. Optional: To export as a `JSON` file, click *Export*.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-executing-plan-proc.adoc]: This is an optional for both steps.

File: doc-content/enterprise-only/processes/process-instance-migration-executing-plan-proc.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
+[id='process-instance-migration-executing-plan-proc']
+= Executing a migration plan
+
+You can execute the process instance migration plan.
+
+.Prerequisites
+* You have defined processes in a backuped {PRODUCT} development environment.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-executing-plan-proc.adoc]: Done

File: doc-content/enterprise-only/processes/process-instance-migration-editing-plan-proc.adoc [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+----
+localhost:3000
+----
+. Log in to the Process Instance Migration service.
+. On the *Process Instance Migration* page, select the *Edit Migration Plan* image:processes/pim-edit-icon.png[] icon on the row of the migration plan you want to edit. The *Edit Migration Plan* window opens.
+. On each tab, modify the details you want to change.
+. Click *Next*.
+. Optional: To export as a `JSON` file, click *Export*.
+. In the *Review & Submit* tab, review the plan and click *Submit Plan*.
+. Optional: To export as a `JSON` file, click *Export*.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-editing-plan-proc.adoc]: This is an optional for both steps.

File: doc-content/enterprise-only/processes/process-instance-migration-editing-plan-proc.adoc [hunk 9]
```diff
@@ -0,0 +1,23 @@
+[id='process-instance-migration-editing-plan-proc']
+= Editing a migration plan
+
+You can edit a process instance migration plan.
+
+.Prerequisites
+* You have defined processes in a backuped {PRODUCT} development environment.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-editing-plan-proc.adoc]: Done

File: doc-content/enterprise-only/processes/process-instance-migration-deleting-plan-proc.adoc [hunk 10]
```diff
@@ -0,0 +1,18 @@
+[id='process-instance-migration-deleting-plan-proc']
+= Deleting a migration plan
+
+You can delete a process instance migration plan.
+
+.Prerequisites
+* You have defined processes in a backuped {PRODUCT} development environment.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-deleting-plan-proc.adoc]: Done

File: doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+. Optionally, to modify the view in the diagram panes, do any of the following tasks:
+* To select text, select the image:processes/pim-selection-icon.png[] icon.
+* To pan, select the image:processes/pim-pan-icon.png[] icon.
+* To zoom in, select the image:processes/pim-zoom-in-icon.png[] icon.
+* To zoom out, select the image:processes/pim-zoom-out-icon.png[] icon.
+* To fit to viewer, select the image:processes/pim-pan-icon.png[] icon.
+. Click *Next*.
+. Optional: To export as a `JSON` file, click *Export*.
+. In the *Review & Submit* tab, review the plan and click *Submit Plan*.
+. Optional: To export as a `JSON` file, click *Export*.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc]: This is an optional for both steps.

File: doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+. In the *Target ContainerID* field, update the target container ID.
+. Click *Retrieve Definition from backend*, and click *Next*.
++
+image::processes/pim-migration-wizard.png[align="center"]
+
+. From the *Source Nodes* list, select the source node you want to map.
+. From the *Target Nodes* list, select the target node you want to map.
+. If the *Source Process Definition Diagram* pane is not displayed, click *Show Source Diagram*.
+. If the *Target Process Definition Diagram* pane is not displayed, click *Show Target Diagram*.
+. Optionally, to modify the view in the diagram panes, do any of the following tasks:
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc]: Done

File: doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+. In the upper right corner of the *Process Instance Migration* page, from the KIE Service list select the KIE Service you want to add a migration plan for.
+. Click *Add Plan*. The *Add Migration Plan Wizard* window opens.
+. In the *Name* field, enter a name for the migration plan.
+. Optional: In the *Description* field, enter a description for the migration plan.
+. Click *Next*.
+. In the *Source ContainerID* field, enter the source container ID.
+. In the *Source ProcessId* field, enter the source process ID.
+. Click *Copy Source To Target*.
+. In the *Target ContainerID* field, update the target container ID.
+. Click *Retrieve Definition from backend*, and click *Next*.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc]: Done

File: doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+* You have defined processes in a backuped {PRODUCT} development environment.
+* The process instance migration service is running.
+
+.Procedure
+. Open the following address in a web browser:
++
+----
+localhost:3000
+----
+. Log in to the Process Instance Migration service.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc]: One is the name of the service and the other is the topic. What do you suggest to resolve this?

File: doc-content/enterprise-only/processes/process-instance-migration-con.adoc [hunk 15]
```diff
@@ -0,0 +1,6 @@
...
+
+For more information about the process instance migration standalone service, see https://github.com/kiegroup/droolsjbpm-integration/tree/master/process-migration-service[Process Instance Migration Service] in https://github.com/kiegroup[_KIE (Drools, OptaPlanner and jBPM )_].
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-con.adoc]: Ok, I will adopt PIM throughout

File: doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc [hunk 16]
```diff
@@ -0,0 +1,44 @@
+[id='process-instance-migration-creating-plan-proc']
+= Creating a migration plan
+
+You can use the process instance migration service to define the migration between two different process definitions, known as a migration plan.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc]: Reworded

File: doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc (hunk 14 above)
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc]: changing to lowercase and resolving

File: doc-content/enterprise-only/processes/process-instance-migration-deleting-plan-proc.adoc [hunk 17]
```diff
@@ -0,0 +1,18 @@
+[id='process-instance-migration-deleting-plan-proc']
+= Deleting a migration plan
+
+You can delete a process instance migration plan.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-deleting-plan-proc.adoc]: The migration plan is for process instance migration. SME want it referred to as "migration plan", and the UI refers to "migration plan"

File: doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+. In the *Source ContainerID* field, enter the source container ID.
+. In the *Source ProcessId* field, enter the source process ID.
+. Click *Copy Source To Target*.
+. In the *Target ContainerID* field, update the target container ID.
+. Click *Retrieve Definition from backend* and click *Next*.
++
+image::processes/pim-migration-wizard.png[align="center"]
+
+. From the *Source Nodes* list, select the source node you want to map.
+. From the *Target Nodes* list, select the target node you want to map.
```
Comment (jstastny-cz) [inline doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc]: If I recall correctly, there's one more step after selecting sth like "Map these two nodes" button.

File: doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc [hunk 19]
```diff
@@ -0,0 +1,44 @@
+[id='process-instance-migration-installing-service-proc']
+= Installing the process instance migration service
```
Comment (jstastny-cz) [inline doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc]: We need to change this chapter. The procedure below applies to building from community bits, that's not what users should do (if they don't insist). I'll provide the steps shortly.

Comment (jstastny-cz) [reply doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc]: 1. Download rhpam-7.4.0-add-ons.zip from CSP.
2. Unzip following entry - rhpam-7.4-process-migration-service-standalone.jar to arbitrary location
3. The service requires kieserver configuration, provided in a yaml file (on top of thorntail specific config):
```config.yml
kieservers:
  - host: http://localhost:8080/kie-server/services/rest/server
    username: kieserver
    password: kieserver123*
  - host: http://localhost:8280/kie-server/services/rest/server
    username: kieserver
    password: kieserver1!
```
example contents of such file can then be:
```config.yml
thorntail:
  deployment:
    process-migration.war:
      jaxrs:
        application-path: /rest
      web:
        login-config:
          auth-method: BASIC
          security-domain: pim
        security-constraints:
          - url-pattern: /*
            roles: [ admin ]
          - url-pattern: /health/*
  datasources:
    data-sources:
      pimDS:
        driver-name: h2
        connection-url: jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE
        user-name: sa
        password: sa
  security:
    security-domains:
      pim:
        classic-authentication:
          login-modules:
            UsersRoles:
              code: UsersRoles
              flag: required
              module-options:
                usersProperties: application-users.properties
                rolesProperties: application-roles.properties
kieservers:
  - host: http://localhost:8080/kie-server/services/rest/server
    username: kieserver
    password: kieserver123*
  - host: http://localhost:8280/kie-server/services/rest/server
    username: kieserver
    password: kieserver1!
```
4. startup the service by:
```
java -jar rhpam-7.4-process-migration-service-standalone.jar -s./config.yml
```
or when need to run at a port offset, add following option `-Dswarm.network.socket-binding-groups.standard-sockets.port-offset=200`

5. By default service is started on localhost:8080, may vary based on your port offset

Comment (jstastny-cz) [reply doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc]: please do change the ports across the docs, 8080 is default when running like this, not 3300 as currently mentioned.

Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc]: @jstastny-cz Is the "-process-migration-service-standalone" file ".jar" or ".zip"?

Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc]: Done.

File: doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc (hunk 18 above)
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-creating-plan-proc.adoc]: Good catch, checked the UI and adopted as:
. Click *Map these two nodes*.

File: doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+    username: KIESERVER_USERNAME
+    password: KIESERVER_PASSWORD
+  - host: http://localhost:8280/kie-server/services/rest/server
+    username: KIESERVER_USERNAME
+    password: KIESERVER_PASSWORD1
+----
+. Start the PIM service:
++
+----
+$ java -jar rhpam-7.4-process-migration-service-standalone.jar -s./config.yml
```
Comment (jstastny-cz) [inline doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc]: can we parameterize the name here also? In the hints above we have \`{PRODUCT_FILE}-process-migration-service-standalone.jar\`

Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-installing-service-proc.adoc]: Good idea, I'll do that

File: doc-content/enterprise-only/processes/process-instance-migration-editing-plan-proc.adoc [hunk 21]
```diff
@@ -0,0 +1,23 @@
+[id='process-instance-migration-editing-plan-proc']
+= Editing a migration plan
+
+You can edit a process instance migration plan.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-editing-plan-proc.adoc]: Added some additional info. I will expand on this further post GA.

File: doc-content/enterprise-only/processes/process-instance-migration-executing-plan-proc.adoc [hunk 22]
```diff
@@ -0,0 +1,27 @@
+[id='process-instance-migration-executing-plan-proc']
+= Executing a migration plan
+
+You can execute the process instance migration plan.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-executing-plan-proc.adoc]: Will expand on use cases for this tool post GA.

File: doc-content/enterprise-only/processes/process-instance-migration-exporting-plan-proc.adoc [hunk 23]
```diff
@@ -0,0 +1,18 @@
+[id='process-instance-migration-exporting-plan-proc']
+= Exporting a migration plan
+
+You can export the process instance migration plan.
```
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/enterprise-only/processes/process-instance-migration-exporting-plan-proc.adoc]: Will add to this post GA

**Pull request merged**